internal/services/contracts_events_consumer_test.go

```diff
@@ -339,6 +344,122 @@ func Test_Ignore_Transfer_Unit_Not_Found(t *testing.T) {
 	s.assert.EqualError(err, "record not found as this might be a newly minted device")
 }
 
+type beneficiaryCase struct {
+	Name                      string
+	Address                   common.Address
+	Event                     ev
+	AutopiUnitTable           models.AutopiUnit
+	ExpectedBeneficiaryResult null.Bytes
+}
+type ev struct {
+	IdProxyAddress common.Address
+	NodeId         *big.Int
+	Beneficiary    common.Address
+}
+
+func TestSetBeneficiary(t *testing.T) {
+	s := initCEventsTestHelper(t)
+	defer s.destroy()
+
+	cases := []beneficiaryCase{
+		{
+			Name:    "Ignore other contracts",
+			Address: common.BigToAddress(big.NewInt(2)),
+			Event: ev{
+				IdProxyAddress: common.BigToAddress(big.NewInt(2)),
+				NodeId:         big.NewInt(2),
+				Beneficiary:    common.BigToAddress(big.NewInt(2)),
+			},
+			AutopiUnitTable: models.AutopiUnit{
+				OwnerAddress: null.BytesFrom(common.BigToAddress(big.NewInt(2)).Bytes()),
+				TokenID:      types.NewNullDecimal(new(decimal.Big).SetBigMantScale(big.NewInt(2), 0)),
+			},
+			ExpectedBeneficiaryResult: null.Bytes{Bytes: []byte{}},
+		},
+		{
+			Name:    "Go from null to explicitly set beneficiary",
+			Address: common.HexToAddress(s.settings.DIMORegistryAddr),
+			Event: ev{
+				IdProxyAddress: common.HexToAddress(s.settings.AftermarketDeviceContractAddress),
+				NodeId:         big.NewInt(1),
+				Beneficiary:    common.BigToAddress(big.NewInt(1)),
+			},
+			AutopiUnitTable: models.AutopiUnit{
+				OwnerAddress: null.BytesFrom(common.BigToAddress(big.NewInt(1)).Bytes()),
+				TokenID:      types.NewNullDecimal(new(decimal.Big).SetBigMantScale(big.NewInt(1), 0)),
+			},
+			ExpectedBeneficiaryResult: null.BytesFrom(common.BigToAddress(big.NewInt(1)).Bytes()),
+		},
+		{
+			Name:    "Go from one explicitly set beneficiary to another",
+			Address: common.HexToAddress(s.settings.DIMORegistryAddr),
+			Event: ev{
+				IdProxyAddress: common.HexToAddress(s.settings.AftermarketDeviceContractAddress),
+				NodeId:         big.NewInt(3),
+				Beneficiary:    common.BigToAddress(big.NewInt(3)),
+			},
+			AutopiUnitTable: models.AutopiUnit{
+				OwnerAddress: null.BytesFrom(common.BigToAddress(big.NewInt(1)).Bytes()),
+				TokenID:      types.NewNullDecimal(new(decimal.Big).SetBigMantScale(big.NewInt(3), 0)),
+				Beneficiary:  null.BytesFrom(common.BigToAddress(big.NewInt(2)).Bytes()),
+			},
+			ExpectedBeneficiaryResult: null.BytesFrom(common.BigToAddress(big.NewInt(3)).Bytes()),
+		},
+		{
+			Name:    "Go from beneficiary to explicitly cleared beneficiary",
```

Comment:
I think the one thing we should probably check still is: I have a unit, the beneficiary is set, then I get a transfer event—do I clear this thing correctly?